hadoop-ozone/client/src/test/java/org/apache/hadoop/ozone/client/rpc/RpcClientTest.java

```diff
@@ -0,0 +1,197 @@
+    NULL_EXPECTED_ONE_OM(
+        null,
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        null,
+        true),
+    NULL_EXPECTED_TWO_OM(
+        null,
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        true),
+    NULL_EXPECTED_TWO_OM_SECOND_MISMATCH(
+        null,
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        "invalid",
+        true),
+    NULL_EXPECTED_TWO_OM_BOTH_MISMATCH(
+        null,
+        "invalid",
+        "invalid",
+        true),
+    NULL_EXPECTED_TWO_OM_FIRST_MISMATCH(
+        null,
+        "invalid",
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        true),
+    EMPTY_EXPECTED_NO_OM(
+        "",
+        null,
+        null,
+        true),
+    EMPTY_EXPECTED_ONE_OM(
+        "",
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        null,
+        true),
+    EMPTY_EXPECTED_TWO_OM(
+        "",
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        true),
+    EMPTY_EXPECTED_TWO_OM_MISMATCH(
+        "",
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        "invalid",
+        true),
+    EMPTY_EXPECTED_TWO_OM_BOTH_MISMATCH(
+        "",
+        "invalid",
+        "invalid",
+        true),
+    VALID_EXPECTED_NO_OM(
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        null,
+        null,
+        false),
+    VALID_EXPECTED_ONE_OM(
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        null,
+        true),
+    VALID_EXPECTED_TWO_OM(
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        true),
+    VALID_EXPECTED_TWO_OM_SECOND_LOWER(
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        "1.0.1",
+        false),
+    VALID_EXPECTED_TWO_OM_SECOND_HIGHER(
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        "100.0.1",
+        true),
+    VALID_EXPECTED_TWO_OM_FIRST_OM_LOWER(
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        "1.0.1",
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        false),
+    VALID_EXPECTED_TWO_OM_FIRST_OM_HIGHER(
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        "10.0.1",
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        true),
+    VALID_EXPECTED_TWO_OM_BOTH_LOWER(
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        "1.0.1",
+        "1.9.1",
+        false),
+    VALID_EXPECTED_ONE_OM_HIGHER_VERSION(
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        "100.0.1",
+        null,
+        true),
+    VALID_EXPECTED_TWO_OM_HIGHER_VERSION(
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        "100.0.1",
+        "100.0.1",
+        true),
+    VALID_EXPECTED_ONE_OM_LOWER_VERSION(
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        "1.0.1",
+        null,
+        false),
+    VALID_EXPECTED_TWO_OM_LOWER_VERSION(
+        OZONE_OM_CLIENT_PROTOCOL_VERSION,
+        "1.0.1",
+        "1.0.1",
```

Comment:
One test case needed here is
OM version return "" as proto default if no value is ""
That would test scenario of newer S3G/RpcClient older OM.

Reply:
Will add them